OpenEdXMobile/src/main/java/org/edx/mobile/view/LoginActivity.java

```diff
@@ -235,24 +235,38 @@ public String getEmail() {
         return activityLoginBinding.emailEt.getText().toString().trim();
     }
 
-    private static final int RESET_PASSWORD_REQUEST_CODE = 0;
-
     private void showResetPasswordDialog() {
-        startActivityForResult(ResetPasswordActivity.newIntent(getEmail()), RESET_PASSWORD_REQUEST_CODE);
+        resetPasswordAlert = ResetPasswordAlertDialogFragment.newInstance(this, getEmail());
+        resetPasswordAlert.setListener(this);
+        resetPasswordAlert.show(getSupportFragmentManager(), null);
     }
 
-    public void showEulaDialog() {
-        environment.getRouter().showWebViewDialog(this, getString(R.string.eula_file_link), getString(R.string.end_user_title));
+    @Override
+    public void onResult(boolean success, @Nullable String errorMessage) {
```

Comment:
Success can be inferred from the presence of the error message, so technically only one parameter is needed here.